src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ArrayCreationAndInitializer.cs

```diff
@@ -0,0 +1,415 @@
+class C
+{
+    public void F()
+    {
+        var a = /*<bind>*/new string[1]/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IArrayCreationExpression (Dimension sizes: 1, Element Type: System.String) (OperationKind.ArrayCreationExpression, Type: System.String[])
+  ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+";
+            VerifyOperationTreeForTest<ArrayCreationExpressionSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")]
+        public void SimpleArrayCreation_UserDefinedType()
+        {
+            string source = @"
+class M { }
+
+class C
+{
+    public void F()
+    {
+        var a = /*<bind>*/new M[1]/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M[])
+  ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+";
+            VerifyOperationTreeForTest<ArrayCreationExpressionSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")]
+        public void SimpleArrayCreation_ConstantDimension()
+        {
+            string source = @"
+class M { }
+
+class C
+{
+    public void F()
+    {
+        const int dimension = 1;
+        var a = /*<bind>*/new M[dimension]/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M[])
+  ILocalReferenceExpression: dimension (OperationKind.LocalReferenceExpression, Type: System.Int32, Constant: 1)
+";
+            VerifyOperationTreeForTest<ArrayCreationExpressionSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")]
+        public void SimpleArrayCreation_NonConstantDimension()
+        {
+            string source = @"
+class M { }
+
+class C
+{
+    public void F(int dimension)
+    {
+        var a = /*<bind>*/new M[dimension]/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M[])
+  IParameterReferenceExpression: dimension (OperationKind.ParameterReferenceExpression, Type: System.Int32)
+";
+            VerifyOperationTreeForTest<ArrayCreationExpressionSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")]
+        public void SimpleArrayCreation_DimensionWithImplicitConversion()
+        {
+            string source = @"
+class M { }
+
+class C
+{
+    public void F(char dimension)
+    {
+        var a = /*<bind>*/new M[dimension]/*</bind>*/;
```

Comment:
Can you also include some error tests with dimensions that are not implicitly convertible, but are used as implicitly convertible anyway?